### PR TITLE
[TESTERS NEEDED] SPU: Operating system LR memory signals

### PR DESCRIPTION
**Pull request by elad335**
Implement operating system based LR memory signals, this should increase performance.
For testers: this pr is rebased before the build of #16466 at the moment, so use the build before for comparison.

Demon Souls (approx 16% FPS increase):
Without optimization (44.4 fps):

image

With optimization (51.6 fps):
image

### CODE REVIEW

File: rpcs3/Emu/Cell/SPUThread.cpp [hunk 1]
```diff
@@ -1214,7 +1229,7 @@ void spu_thread::dump_regs(std::string& ret, std::any& /*custom_data*/) const
 			is_const = false;
 		}
 
-		fmt::append(ret, "%s%s ", spu_reg_name[i], is_const ? "©" : ":");
+		fmt::append(ret, "%s%s ", spu_reg_name[i], is_const ? "Â©" : ":");
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/SPUThread.cpp:1232]: Please change your IDE's encoding.

**Comment by Linear524** [comment]
Sounds very cool !)
What cases should we test ?
Or what titles ?

**Comment by elad335** [comment]
Demon Souls (~10% FPS increase):
Without optimization:

image

With optimization:
image

**Comment by elad335** [comment]
> Sounds very cool !) What cases should we test ? Or what titles ?

I have no idea, perhaps games with low FPS but also low CPU usage.

**Comment by digant73** [comment]
@elad335 could you merge in your PR the last merged PR?

**Comment by elad335** [comment]
Managed to squeeze a few more frames!

**Comment by Linear524** [comment]
MGS4 = +4~7 fps
Journey = +2 fps
Killzone3 = +9 fps
CPU - Intel Xeon 2690v4, 32GB Ram, GeForce 1080
P.S. - I'm still using TSX option, is there any benefits to continue using it ? Or this path is dead now ?

**Comment by elad335** [comment]
Thanks for testing, also provide the 'before' fps so we can see the performance gain in relation to master. Also lol, technically for TSX the results would be worse because of all the memory dereferencing so I considered disabling it for that. But because you reported you have TSX it reassured me that I don't need to disable it for TSX. 👍

**Comment by digant73** [comment]
> MGS4 = +4~7 fps Journey = +2 fps Killzone3 = +9 fps CPU - Intel Xeon 2690v4, 32GB Ram, GeForce 1080 P.S. - I'm still using TSX option, is there any benefits to continue using it ? Or this path is dead now ?

For Killzone 3, could you please tell me some sections of the game where you see the fps increase?

**Pull request merged**